src/Documents/Session/Loaders/QueryIncludeBuilder.php

```diff
@@ -56,58 +60,52 @@ public function includeDocuments(?string $path): QueryIncludeBuilder
-//        _includeTimeSeriesFromTo("", name, from, to);
-//        return this;
-//    }
-//
-//    public function includeTimeSeries(?string path, ?string name): QueryIncludeBuilder {
-//        return includeTimeSeries(path, name, null, null);
-//    }
-//
-//    public function includeTimeSeries(?string path, ?string name, Date from, Date to): QueryIncludeBuilder {
-//        _withAlias();
-//        _includeTimeSeriesFromTo(path, name, from, to);
-//        return this;
-//    }
+    public function includeTimeSeries(?string $name, ?DateTimeInterface $from = null, ?DateTimeInterface $to = null): QueryIncludeBuilder
+    {
+        return $this->includeTimeSeriesWithPath('', $name,$from, $to);
+    }
+
+    public function includeTimeSeriesWithPath(?string $path, ?string $name, ?DateTimeInterface $from = null, ?DateTimeInterface $to = null): QueryIncludeBuilder
+    {
+        $this->_withAlias();
+        $this->_includeTimeSeriesFromTo($path, $name, $from, $to);
+        return $this;
+    }
 
     public function includeCompareExchangeValue(?string $path): QueryIncludeBuilder
     {
         $this->_includeCompareExchangeValue($path);
         return $this;
     }
 
-//    public function includeTimeSeries(?string name, TimeSeriesRangeType type, TimeValue time): QueryIncludeBuilder {
-//        _includeTimeSeriesByRangeTypeAndTime("", name, type, time);
-//        return this;
-//    }
-//
-//    public function includeTimeSeries(?string name, TimeSeriesRangeType type, int count): QueryIncludeBuilder {
-//        _includeTimeSeriesByRangeTypeAndCount("", name, type, count);
-//        return this;
-//    }
-//
-//    public function includeTimeSeries(?string[] names, TimeSeriesRangeType type, TimeValue time): QueryIncludeBuilder {
-//        _includeArrayOfTimeSeriesByRangeTypeAndTime(names, type, time);
-//        return this;
-//    }
-//
-//    public function includeTimeSeries(?string[] names, TimeSeriesRangeType type, int count): QueryIncludeBuilder {
-//        _includeArrayOfTimeSeriesByRangeTypeAndCount(names, type, count);
-//        return this;
-//    }
-//
-//    public function includeAllTimeSeries(TimeSeriesRangeType type, TimeValue time): QueryIncludeBuilder {
-//        _includeTimeSeriesByRangeTypeAndTime("", Constants.TimeSeries.ALL, type, time);
-//        return this;
-//    }
-//
-//    public function includeAllTimeSeries(TimeSeriesRangeType type, int count): QueryIncludeBuilder {
-//        _includeTimeSeriesByRangeTypeAndCount("", Constants.TimeSeries.ALL, type, count);
-//        return this;
-//    }
+    public function includeTimeSeriesRangeType(null | string | array $names, TimeSeriesRangeType $type, TimeValue | int $timeOrCount): QueryIncludeBuilder
+    {
+        if (is_string($names)) {
+            if (is_int($timeOrCount)) {
+//                $this->_includeTimeSeriesByRangeTypeAndCount("", $names, $type, $timeOrCount);
```

Comment:
what about that?

Reply:
none of the test required it yet, it just have to be uncommented. base method is implemented

Reply:
ok sure. :)